inference/worker/requirements.txt

```diff
@@ -1,12 +1,15 @@
 hf_transfer
 huggingface_hub
+langchain==0.0.142
 loguru
 lorem
+numexpr
```

Comment:
would this maybe mean that we would need to depend on whatever pypi libraries needed for the "default" or "standard" plugins to work and be delivered with OA (e.g. even if user did not want to use them).

e.g. project would now be depending on `numexpr` just for the calculator plugin. 

not sure if i'm understanding it correct or not but just wanted to double check.

i also wonder if anything we can learn from how other project approach this - Airflow could be a good example as they have been doing plugins for years so maybe have some good lessons learned too. https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html

for example/reference here is an anomaly detection plugin/provider i have been working on recently for airflow: https://github.com/andrewm4894/airflow-provider-anomaly-detection

at the same time maybe stuff like this is things to be considered later once we all understand how we'd like it all to work would be.

Reply:
Hi, 
the plugin is included just as an example, they could easily be separate repo or something like that, also if OA decides to provide some default plugins that are not user-provided, e.g. hosted from OA side, they can be written in any language/framework as they are just independent things/API-wrappers.

You can checkout some of the available plugins at PLUGINS.md file, a made some references.

Reply:
I think even for just having as an example it would make sense to have a separate `requirements.txt` in each `plugins/[...]` subdirectory, so that once this is merged to main the worker image won't have the additional dependencies installed for no reason